### PR TITLE
Use existing cmake file where present

### PR DESCRIPTION
**Pull request by thomas-burley-bluefruit**

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -9,16 +9,38 @@ IFS=';' read -ra paths <<< "$1"
...
+  if [ -z "$2" ]; then
+    source_dir="."
+  else
+    source_dir=$2
+  fi
 
   # generate a coverage report
   uuid=$(uuidgen)
-  gcovr --branches --xml-pretty -r . >> "$uuid-report.xml"
+  gcovr . --root "$source_dir"  --branches --xml-pretty >> "$uuid-report.xml"
```
Comment (grahame-student) [inline entrypoint.sh:45]: gcovr is being given 2 paths, ```.``` and ```"$source_dir```, do we know what gcovr will do with the 2 paths?

note that previously ```.``` was being passed as the root path

File: .github/workflows/test_released_action.yml [hunk 2]
```diff
@@ -12,7 +12,12 @@ jobs:
...
-        uses: apollo-fire/gtest-action@v0.0.3
+      - name: GTest Action
+        uses: apollo-fire/gtest-action@v0.0.4
         with:
           test-path: 'test/working_unittest_project'
+
+      - name: GTest Action with existing CMake file
+        uses: apollo-fire/gtest-action@v0.0.4
+        with:
+          test-path: 'test/working_unittest_project_with_existing_cmake'
```
Comment (grahame-student) [inline .github/workflows/test_released_action.yml:23]: Might be worth also testing a hybrid usage, there could be a scenario where both are in use if Apollo opt to transition to the new version in the future.
```suggestion
      - name: GTest Action with existing CMake file
        uses: apollo-fire/gtest-action@v0.0.4
        with:
          test-path: 'test/working_unittest_project_with_existing_cmake'

      - name: GTest Action with mix of projects with and without existing CMake file
        uses: apollo-fire/gtest-action@v0.0.4
        with:
          test-path: "test/working_unittest_project;\
            test/working_unittest_project_with_existing_cmake"
```

Careful with the quotes in the suggestion, changing the double quotes to singles could impact line wrapping by introducing spaces where they are not intended!

**Review by grahame-student — COMMENTED** [review]
Looking good, I've added a few questions / observations

File: entrypoint.sh (hunk 1 above)
Comment (thomas-burley-bluefruit) [reply entrypoint.sh:45]: In this instance, `.` is what gcovr refers to as the *search_paths*, which is where the coverage files are expected to be. `"$source_dir"` is referring to the location of the source files which we want coverage reports for. As per the [gcovr CLI ref](https://gcovr.com/en/master/manpage.html#cmdoption-gcovr-r):

> -r <root>, --root <root>
> The root directory of your source files. Defaults to ‘.’, the current directory. File names are reported relative to this root. The --root is the default --filter. Config key(s): root.

With our configuration of having source files in a different directory to test files, and then targeting them via CMake library, gcovr wasn't able to see the source files it was supposed to be generating reports for. This meant I was just getting information on the tests themselves as well as some gtest code!

File: .github/workflows/test_released_action.yml (hunk 2 above)
Comment (thomas-burley-bluefruit) [reply .github/workflows/test_released_action.yml:23]: Good idea! I'll implement that now.

**Comment by thomas-burley-bluefruit** [comment]
> Looking good, I've added a few questions / observations

Ready for re-review.

**Comment by grahame-student** [comment]
Just spotted that the readme.md needs updating to show the new version and document the new param. With that small change I'll be happy to approve.

File: entrypoint.sh (hunk 1 above)
Comment (grahame-student) [reply entrypoint.sh:45]: That makes sense, thanks.

**Comment by thomas-burley-bluefruit** [comment]
> Just spotted that the readme.md needs updating to show the new version and document the new param. With that small change I'll be happy to approve.

Done.

**Review by grahame-student — APPROVED** [review]
LGTM

**Review by ThatTallGuy14 — APPROVED** [review]

**Review by Treborlord — APPROVED** [review]

**Pull request merged**